### PR TITLE
Event variants

### PR DESCRIPTION
**Pull request by hugihlynsson**
Adds a page for event variants at /data-design/avo-tracking-plan/event-variants and a mention to the event docs. It would be great to get a review on the copy, screenshots and the location of the new page.

### CODE REVIEW

**Comment by hugihlynsson** [comment]
#### 📝 Changed routes:


| [**…ng-plan/event-variants**](https://docs-14js93d5i.avo-dev.app/data-design/avo-tracking-plan/event-variants) | [**…o-tracking-plan/events**](https://docs-14js93d5i.avo-dev.app/data-design/avo-tracking-plan/events) |
|:-:|:-:|
|<a href="https://docs-14js93d5i.avo-dev.app/data-design/avo-tracking-plan/event-variants">Screenshot of /data-design/avo-tracking-plan/event-variants</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/5XA9YVmhKkxkne5wAtcXn6OKQ1V4-hcs7Z8_-s9OUZJDNfAUK1ixuwfEEHQRX_6w0VdwAzG6lWo_bxOyI-crAg.png">(view full size)</a> |<a href="https://docs-14js93d5i.avo-dev.app/data-design/avo-tracking-plan/events">Screenshot of /data-design/avo-tracking-plan/events</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/KD3uLVXLdwjBoMY64u-McZf8Mvm8tX3EBLSHX283Gt3TZk6Rjzp1lKKSmk14Zl2F-JTxmSOnKQACS_HjyqqF-A.png">(view full size)</a> |







Commit 74f5e9011d7d7ec938c45fe709b2b5ac6ce54188 (https://docs-14js93d5i.avo-dev.app).

**Comment by hugihlynsson** [comment]
> Weird, but it shows the `Event Variants` in the left panel only if I open the `Events` in the left panel 😄
Good catch! I've fixed this

File: pages/data-design/avo-tracking-plan/event-variants.mdx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+Event variants streamline the tracking plan by allowing variations of an event to suit different scenarios, making the specification more precise. Each variant, inheriting properties from a base event, can be modified and enhanced. This approach simplifies both implementation and communication with developers, as it narrows the scope of changes to only necessary areas. Additionally, event variants generate their own Codegen functions, facilitating quick and accurate implementation by developers and reducing uncertainty about property values. Existing events with numerous conditional properties used across various scenarios can greatly benefit from event variants.
```
[the remark this answers is not in the document]
Comment (hugihlynsson) [reply pages/data-design/avo-tracking-plan/event-variants.mdx:7]: Great idea. I've added an example but we should probably break this further down and be more concrete with the value of event variants

File: pages/data-design/avo-tracking-plan/event-variants.mdx [hunk 2]
```diff
@@ -0,0 +1,77 @@
+import { Callout } from 'nextra/components'
+
+# Event variants
+
+Event variants are a powerful feature that allows you to create variations of an event to suit different scenarios. This makes the tracking plan more precise, reduces the scope of tracking changes and simplifies Codegen implementation.
+
+## Introduction
+
+When defining a tracking plan for events that are used in multiple scenarios within a product, there can often be a need for making the specifications for an event more granular. For example some properties might need to be included in some scenarios but not others. Property descriptions, event descriptions and triggers can help clarify the data structures needed for each scenario but it still leaves a communication gap, for example when the developer is implementing an event. Making trivial changes to an event that’s implemented in many scenarios can cause widespread changes in the implementation of an event, even if it’s only meant to target a specific scenario. This slows down implementation and makes those designing data hesitant to introduce improvements to the tracking plan.
```
Comment (1212spark) [inline pages/data-design/avo-tracking-plan/event-variants.mdx:9]: "For example," with a comma

File: pages/data-design/avo-tracking-plan/event-variants.mdx [hunk 3]
```diff
@@ -0,0 +1,77 @@
+import { Callout } from 'nextra/components'
+
+# Event variants
+
+Event variants are a powerful feature that allows you to create variations of an event to suit different scenarios. This makes the tracking plan more precise, reduces the scope of tracking changes and simplifies Codegen implementation.
```
[the remark this answers is not in the document]
Comment (hugihlynsson) [reply pages/data-design/avo-tracking-plan/event-variants.mdx:5]: I actually wanted to focus on Codegen here as that's where most of the scoping improvements will be. When using a home-grown analytics solution, there is less of a connection between variants and the implementation.

File: pages/data-design/avo-tracking-plan/event-variants.mdx (hunk 2 above)
Comment (hugihlynsson) [inline pages/data-design/avo-tracking-plan/event-variants.mdx:9]:
```suggestion
When defining a tracking plan for events that are used in multiple scenarios within a product, there can often be a need for making the specifications for an event more granular. For example, some properties might need to be included in some scenarios but not others. Property descriptions, event descriptions and triggers can help clarify the data structures needed for each scenario but it still leaves a communication gap, for example when the developer is implementing an event. Making trivial changes to an event that’s implemented in many scenarios can cause widespread changes in the implementation of an event, even if it’s only meant to target a specific scenario. This slows down implementation and makes those designing data hesitant to introduce improvements to the tracking plan.
```

Comment (hugihlynsson) [reply pages/data-design/avo-tracking-plan/event-variants.mdx]: Added the comma!

**Review by logason — APPROVED** [review]
LGTM!

**Comment by hugihlynsson** [comment]
> Latest vercel preview still does not show "Event Variants" in the menu if I'm not on the "Events" page

Yeah I'm seeing the same, super weird. This doesn't happen locally so I'm guessing that this is some caching gone wrong or something like that.

**Pull request merged**